### DISCUSSION TITLE
Tracing or Metrics For Cache Reads/Writes

### DISCUSSION THREAD

**Discussion by bsulliva** [open-telemetry/opentelemetry-dotnet#4500] [Q&A]
Hi, is there a best practice around tracing items being added/retrieved from cache in .net core apps? Is that something we need to use manual instrumentation to achieve? Or is there any automatic instrumentation available? Or maybe its better to use metrics for this? My end goal is to really just see what kind of cache hit ratio we are seeing. What percentage of time are we pulling items from cache versus going to the source.

Thanks

**Comment by cijothomas** [comment]
If the cache already exposes the stats (hits/miss etc), then it could be reported as metric using the ObervableCounter/Gauge/UpDownCounter. 
Otherwise, you'll have to report it manually using Counter instruments.

**Reply by bsulliva** [reply]
Thanks @cijothomas. Do you have any references to docs that show how to expose the stats for cache? Or any docs on the ObservableCounter/Gauge/UpDownCounter? Just trying to understand how to implement this. Thanks

**Reply by cijothomas** [reply]
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics Please start with the docs from here for learning about the metrics part.
